examples/eight_schools/main_svi.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import absolute_import, division, print_function
+
+import argparse
+import logging
+import sys
+
+import torch
+
+import pyro
+import pyro.distributions as dist
+from data import J, sigma_tensor, y_tensor
+from pyro.infer import SVI, Trace_ELBO
+from pyro.optim import Adam
+
+softplus = torch.nn.Softplus()
+data = torch.stack([y_tensor, sigma_tensor], dim=1)
+
+
+def model(data):
+    y = data[:, 0]
+    sigma = data[:, 1]
+
+    eta_dist = dist.Normal(torch.zeros(J), torch.ones(J))
+    mu_dist = dist.Normal(loc=torch.zeros(1), scale=10 * torch.ones(1))
+    tau_dist = dist.HalfCauchy(loc=torch.zeros(1), scale=25 * torch.ones(1))
+
+    eta = pyro.sample('eta', eta_dist)
+    mu = pyro.sample('mu', mu_dist)
+    tau = pyro.sample('tau', tau_dist)
+
+    theta = mu + tau * eta
+
+    pyro.sample("obs", dist.Normal(theta, sigma), obs=y)
+
+
+def guide(data):
+    eta_loc = torch.randn(J, 1)
+    # note that we initialize our scales to be pretty narrow
+    eta_log_sig = torch.tensor(-3.0 * torch.ones(J, 1) + 0.05 * torch.randn(J, 1))
+    mu_loc = torch.randn(1)
+    mu_log_sig = torch.tensor(-3.0 * torch.ones(1) + 0.05 * torch.randn(1))
+    logtau_loc = torch.randn(1)
+    logtau_log_sig = torch.tensor(-3.0 * torch.ones(1) + 0.05 * torch.randn(1))
+
+    # register learnable params in the param store
+    m_eta_param = pyro.param("guide_mean_eta", eta_loc)
```

Comment:
nit: would suggest changing the names to `loc_eta`, `scale_eta`, etc. for readability.